_posts/2024-06-30-cryostat-release.md

```diff
@@ -0,0 +1,81 @@
+4. **S3-compatible object storage**: previous versions of Cryostat used direct storage of files on disk for archived Flight Recordings and custom Event Templates. Cryostat 3.0 no longer uses direct filesystem accesses for these kinds of binary large objects, but instead integrates with an S3-compatible storage provider. If you use the Cryostat Operator or Cryostat Helm Chart for installation, your deployment will include a [`cryostat-storage`](https://github.com/cryostatio/cryostat-storage) container (a lightly customized [SeaweedFS](https://github.com/seaweedfs/seaweedfs)) for a "batteries included" storage solution. If you roll your own deployment then you can swap this out for any S3-compatible storage provider you prefer or may already use.
+5. **Reverse proxy architecture**: Cryostat 3.0 deployed by the Cryostat Operator or Helm Chart includes a reverse proxy ([`openshift-oauth-proxy`](https://github.com/openshift/oauth-proxy/) or [`oauth2_proxy`](https://github.com/oauth2-proxy/oauth2-proxy)) in the Pod. Only this proxy is exposed to cluster traffic via a Service. This means that all API requests to Cryostat, as well as all visitors to the [Cryostat Web UI](https://github.com/cryostatio/cryostat-web) or [Grafana dashboard](https://github.com/cryostatio/cryostat-grafana-dashboard) flow through this proxy. The proxy handles user sessions to control access to the application, providing unified access control and user sessions for both the Cryostat Web UI and Grafana dashboard. Both of these UIs are accessible via the same Ingress or Route and present the same TLS certificate. When deployed on [OpenShift](https://developers.redhat.com/products/openshift/overview), the auth proxy will use its Namespace (the Cryostat installation Namespace) to perform Role-Based Access Control checks for user authentication and authorization by integrating with the OpenShift cluster SSO provider, very similar to what previous Cryostat releases directly implemented. Additionally, the auth proxy can be optionally configured with an `htpasswd` file to enable Basic authentication. On OpenShift this allows for defining additional user accounts that may access the Cryostat application beyond those with OpenShift SSO RBAC access, whereas on other Kubernetes it is the primary authentication mechanism.
+
+#### Operator
+
+Specifically for the Cryostat Operator there are a few items of note:
+1. **Cluster-wide installation** only: the Cryostat Operator can now only be installed cluster-wide, rather than into a subset of cluster namespaces. Cluster-wide installation is the preferred mode for the Operator Lifecycle Manager and per-namespace installations are a deprecated feature, which our Operator has now dropped.
```

Comment:
> Cluster-wide installation is the preferred mode for the Operator Lifecycle Manager and per-namespace installations are a deprecated feature

Make sense to reference an announcement/discussion/doc for this? I couldn't find one ...

Reply:
I think the best public info I can find is the docs on OLM 1.0 in the OpenShift docs:
https://docs.openshift.com/container-platform/4.15/operators/olm_v1/arch/olmv1-operator-controller.html#olmv1-operator-controller

Particularly the part:
>  In OLM 1.0, Operator objects are cluster-scoped. This differs from earlier OLM versions where Operators could be either namespace-scoped or cluster-scoped, depending on the configuration of their related Subscription and OperatorGroup objects.

Reply:
That should work I think! Thanks for finding that!!